addons/viewrestriction/functions/fnc_switchPreserveView.sqf

```diff
@@ -0,0 +1,50 @@
+/*
+ * Author: Dystopian
+ * Controls View Preserving state.
+ *
+ * Arguments:
+ * 0: Enabled <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * true call acex_viewrestriction_fnc_switchPreserveView
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_enabled"];
+
+if (!_enabled || {GVAR(mode) > 0}) exitWith {
+    if (isNil QGVAR(preserveViewCameraViewEH)) exitWith {};
+    ["cameraView", GVAR(preserveViewCameraViewEH)] call CBA_fnc_removePlayerEventHandler;
+    ["vehicle", GVAR(preserveViewVehicleEH)] call CBA_fnc_removePlayerEventHandler;
+    GVAR(preserveViewCameraViewEH) = nil;
+    GVAR(preserveViewVehicleEH) = nil;
+};
+
+GVAR(preserveViewCameraViewEH) = ["cameraView", {
+    params ["_player", "_cameraView"];
+    if !([_cameraView, cameraOn] call FUNC(canChangeCamera)) exitWith {};
+
+    private _vehicle = vehicle _player;
+    private _vehicleClass = {if (_vehicle isKindOf _x) exitWith {_x}} forEach ["CAManBase", "LandVehicle", "Air", "Ship", "All"];
+    private _varName = QGVAR(preserveView) + _vehicleClass;
```

Comment:
Use `format [QGVAR(preserveView%1), _vehicleClass]`.

Reply:
it's slower not faster

Reply:
I'm sure. It's faster for more than 2 strings. You said that too above https://github.com/acemod/ACEX/pull/107#discussion_r167433555 :laughing:

Reply:
🤕